harmony/utils.py

```diff
@@ -2,24 +2,28 @@
 
 
 def get_batch_codes(batch_mat, batch_key):
-    if type(batch_key) is str or len(batch_key) == 1:
-        if not type(batch_key) is str:
-            batch_key = batch_key[0]
+    if type(batch_key) is str:
```

Comment:
True, for strings of length one this would happen. I prefer unnested branches, but I guess this tradeoff is somewhat of a personal preference. Another way to do this would be

```python
def get_batch_codes(batch_mat, batch_key):
    if isinstance(batch_key, (str, list)) and len(batch_key) == 1:
        batch_key = batch_key if isinstance(batch_key, str) else batch_key[0]
        batch_vec = batch_mat[batch_key]
    else:
        batch_vec = batch_mat[batch_key].astype("str").apply(','.join, axis=1)

    return batch_vec.astype("category")
```

Let me know what you think.

Reply:
@mbruhns In your code above, I'm afraid that if `batch_key` is a string, it will never go into the then block.

Reply:
I may personally prefer keeping the old code in this part if it doesn't affect the results. What do you think?

Reply:
I agree, let's stick to the original code!